src/components/Header.tsx

```diff
@@ -0,0 +1,113 @@
+import { useCallback, useContext, useEffect } from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Col, Container, Nav, Navbar, Row } from 'react-bootstrap'
+import { Link, useLocation } from 'react-router-dom'
+import axios, { AxiosError } from 'axios'
+import { ErrorResponse } from '../types'
+import { useIsLoggedIn, useLogout } from '../utils/hooks'
+import { getTokenLocalStorage } from '../utils/helpers'
+import { FlashMessageContext, FlashMessageType } from '../context/flashmessage'
+import { ProfileContext } from '../context/profile'
+import { TokenContext } from '../context/token'
+import { getProfile } from '../services/api'
+import useTranslationHook from '../hook/TranslationHook'
+import PageHeader from './PageHeader'
+import { PAGE_CONFIG } from '../utils/constants'
+
+const Header = (): JSX.Element => {
+  const { token, setToken } = useContext(TokenContext)
+  const { profile, setProfile } = useContext(ProfileContext)
+  const { addErrorMessage } = useContext(FlashMessageContext)
+  const logout = useLogout()
+  const { t } = useTranslationHook()
+  const location = useLocation()
+  const currentPage = PAGE_CONFIG.find(page => location.pathname.includes(page.path))
+
+  const addErrorMessageCallback = useCallback(
+    (flashMessage: FlashMessageType) => addErrorMessage(flashMessage),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  const handleFetchProfileErrorCallback = useCallback((error: unknown | AxiosError): void => {
+    console.error(error)
+    if (axios.isAxiosError(error)) {
+      const axiosError = error as AxiosError<ErrorResponse>
+      const { response } = axiosError
+      if ((response?.status === 401) && (response.data?.detail === 'Invalid token.')) {
+        addErrorMessageCallback({ body: 'Session expired, please log in again.' })
+        logout()
+      }
+    } else {
+      addErrorMessageCallback({ body: 'Error fetching profile data' })
+    }
+  },
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  [addErrorMessageCallback]
+  )
+
+  const fetchProfileCallback = useCallback(async (token: string): Promise<void> => {
+    try {
+      const data = await getProfile(token)
+      setProfile(data)
+    } catch (error: unknown | AxiosError) {
+      handleFetchProfileErrorCallback(error)
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [addErrorMessageCallback, handleFetchProfileErrorCallback])
+
+  // fetch profile
+  useEffect(() => {
+    if (token === null || token === undefined) return
+    // eslint-disable-next-line no-void
+    void (async () => await fetchProfileCallback(token))()
+  }, [token, fetchProfileCallback])
+
+  useEffect(() => {
+    setToken(getTokenLocalStorage())
+  }, [setToken])
+
+  return (
+    <>
+      {useIsLoggedIn() === true && (
+        <Navbar className='shadow-sm mb-5' fixed='top' bg='body' data-bs-theme='light'>
+          <Container fluid>
```

Comment:
Maybe whatever is inside that block could go into a LoggedInHeader component

Reply:
add LoggedInHeader component for not duplicate some code between footer and header

Reply:
I meant a component not a hook. It's cool to have a hook now, but we also need a component